numpy/lib/tests/test_format.py

```diff
@@ -934,6 +936,26 @@ def test_unicode_field_names(tmpdir):
         with assert_warns(UserWarning):
             format.write_array(f, arr, version=None)
 
+def test_header_growth_axis():
+    import io
+    for is_fortran_array, dtype_space, expected_header_length in [
+        [False, 27, 128],
+        [False, 28, 192],
+        [True, 31, 128],
+        [True, 32, 192]
+    ]:
+        dtype = [(' '*dtype_space, int)]
+        arr = np.arange(2000).astype(dtype).reshape(2, 1000)
+
+        if is_fortran_array:
+            arr = np.asfortranarray(arr)
+
+        fp = io.BytesIO()
+        format.write_array_header_1_0(
+            fp, format.header_data_from_array_1_0(arr)
```

Comment:
I would recommend constructing test dicts directly instead of grabbing them from a real array. Then you will be more able to test arbitrary lengths of the growth axis, and thus test the important property of the feature introduced here: the invariance of the header length under such changes.

Reply:
That's is a great idea, just implemented this.